tf-ncl/src/nickel_builder.rs

```diff
@@ -17,77 +19,94 @@ pub struct Complete(Option<RichTerm>);
 pub struct Field<RB> {
     record: RB,
     path: StaticPath,
-    metadata: Option<MetaValue>,
+    metadata: FieldMetadata,
+}
+
+pub struct Types(pub TypeF<Box<Types>, RecordRows, EnumRows>);
+
+impl From<Types> for TypeF<Box<Types>, RecordRows, EnumRows> {
+    fn from(value: Types) -> Self {
+        value.0
+    }
+}
+
+fn add_positions(t: Types) -> types::Types {
```

Comment:
The naming is a bit surprising, because it seems this function actually _clears_ the positions? Or is it because your representation is position-less? Then, would implementing a `From` trait make more sense?